docs/development.md

```diff
@@ -0,0 +1,142 @@
+ 
+ You can find more information about the data flow [here](http://redux.js.org/docs/basics/DataFlow.html)
+ ### Server side architecture
+ //TODO 
+
+ 
+## Project's structure
+
+ 
+ The client-side javascript sits inside the assets folder and it is organized in the following way:
+ 
+```
+ .
+ +-- assets
+ |   +-- components  // Shared components.
+ |   |   +-- exampleComponent1
+ |   |   |   +-- exampleComponent.jsx
+ |   |   |   +-- exampleComponent.css
+ |   |   +-- exampleComponent2
+ |   +-- helpers     // Shared helpers.
+ |   |   +-- helper.js  
+ |   +-- images      
+ |   +-- pages       // Support's pages
+ |   |   +-- page1   // An specific page.    
+ |   |   |   +-- actions    // Actions of a certain page/redux app.
+ |   |   |   +-- components // Components of a certain page/redux app.
+ |   |   |   |   +-- component1.jsx
+ |   |   |   |   +-- component2.jsx
+ |   |   |   +-- helpers    // Helpers of a certain page/redux app.
+ |   |   |   +-- reducers   // Reducers of a certain page/redux app. 
+ |   |   |   +-- page1.jsx  // Root component of a page.
+ |   |   |   +-- page1.scss // Stylesheet containing all the style of this page's components.
+ |   |   +-- page2
+ |   +-- stylesheets // Shared stylesheets
+ |       +-- main.scss //Entry point of the scss files.
+ ...
+```
+
+#### Important notes:
+
+* The UI of the project is organized in components. The shared components are self-contained and are located in the top
+ `components` folder. The components specific to a page and which are not used in other pages are located inside the 
+ `components` folder inside a specific page.
+ 
+* The CSS for a non-shareable component is located inside the `page.scss` file.  
+ 
+
+## Building process
+
+In order to build the project, team city runs a series of steps. The first step installs node js, the second build the 
+assets by executing the script [`build-tc`](https://github.com/guardian/support-frontend/blob/master/build-tc). 
+Finally, the third step builds the app by running the `compile` task define in 
+[`build.sbt`](https://github.com/guardian/support-frontend/blob/master/build.sbt).
+
+### Building assets
+
+Our building process uses `npm scripts` as a orchestrator tool for assets. The available scripts are defined in the 
+`package.json`. Those scripts, depending on what is the objective use other tools like [webpack](https://webpack.js.org/), 
+[sass](http://sass-lang.com/guide), [babel](https://babeljs.io/), [eslint](http://eslint.org/) etc.
+
+As an example, in order to build the assets for production, the step `build-prod` should be ran. This script runs:
+  1. `clean` : Deletes the previous compiled assets.
+  2. `validate` : Validates the javascript source code by running lint for style check and flow type check.
+  3. `test` : Runs the javascript tests using jest.
+  4. `webpack` : Runs webpack in production mode. Webpack runs babel, minify (uglify) the javascript, does the asset 
```

Comment:
might be good to include slightly more detail than just 'asset hashing'.  Eg, what assets are hashed and how does the application link to hashed assets?